### ISSUE TITLE
Rename Moon

### ISSUE THREAD

**Issue by kbrsh** [kbrsh/moon#152]
I've been thinking about doing this for quite a while now, but kept on pushing it farther and farther away. The reality is still inevitable and as much as I'd like to avoid it, Moon needs to be renamed. It'll give me more freedom to expand the community. Renaming Moon also has a number of advantages:

* Smaller domain
* Better SEO and visibility
* Shorter NPM package name
* Unique file extension (`.moon` is used by MoonScript)
* Official Twitter account

Upgrading should be fairly simple.

* All development will continue on a renamed version
* `moonjs` will be kept on NPM for all versions up to `v0.11.0`
* All new versions (including alphas, betas, and RC's) will be put on the renamed version
* The current documentation will redirect to the new one
* This repo will redirect to the new project, which will have a note about this rename
* All existing projects using Moon do not need to take action, but it is recommended

I haven't decided on the name yet, but it should preferably have something to do with a UI library, be four letters, and start with the letter m (to avoid changing directives).

**Comment by LeviSchuck** [comment]
I've been wondering about that, but it is probably a good decision to make before a release quality v1

**Comment by shnbhg** [comment]
I agree with @LeviSchuck

**Comment by LeviSchuck** [comment]
random things that start with M

Maoi 
Marq
Marz
Mati
Meeq
Meav
Meui
Meki 
Meji
Megi 

This is hard.

**Comment by seanseany** [comment]
prevue 😛

**Comment by kbrsh** [comment]
@LeviSchuck How about:

* Mora
* Mova
* Mino

@seanseany I don't think that'll be a good name, because the v1 API is becoming more and more different than both Preact and Vue.

**Comment by LeviSchuck** [comment]
Mino already exists on npmjs 

Mora has an associated user 

I do like Mova

**Comment by Margaret2** [comment]
I like the name Moon, but I can understand you wanting to change it, so here's some more ideas:
Mint 
Mist
Miso
Moka 
Matcha
Mira
Mesa
Maui

**Comment by itsjef** [comment]
I'd suggest the name **Mirai** - which means **the future** in Japanese. The **mi** in **mirai** also means **to view** or **vision** which might be suitable.

Also, another name is **Mochi** - a type of small, usually round-shaped moon-like (see the pun?) Japanese sweet, although [mochijs](https://github.com/StevenPerez/MochiJS) has already been taken but its latest commit was back to 2015 so I think that wouldn't be a problem.

**Comment by LeviSchuck** [comment]
Mochi may be a bad choice as a project already exists under this name for a web technology, albeit backend, https://github.com/mochi/mochiweb

**Comment by LeviSchuck** [comment]
Miso is also a bad choice, as a Haskell to JS project exists under that name, https://github.com/haskell-miso/miso

Mesa is also a bad choice as an open GL implementation is under that name for a long time. For a developer it won't be helpful to search that name.

**Comment by eucariop** [comment]
Blazify

**Comment by LeviSchuck** [comment]
If only there wasn't something "Blaze"-y that involved HTML 😛 

https://hackage.haskell.org/package/blaze-html

**Comment by eucariop** [comment]
Ok, what does BlazifyJS say?

**Comment by itsjef** [comment]
We shouldn't be worried about non-js existing technology/framework except already well-known ones since we can always append **-js** at the end :P

**Comment by kbrsh** [comment]
First of all, I want to thank you all for helping. :heart:

@LeviSchuck Regarding Mino--it's taken on NPM but is being squatted (held with a package that has no functionality). I can easily email the author to get it, or let NPM know and they should transfer it.

I like Mova and Mora so far.

**Comment by PolymathWhiz** [comment]
I like Muto

**Comment by PolymathWhiz** [comment]
What about *Mutod*

**Comment by fengzilong** [comment]
if you guys need mino, I'm glad to hand over the ownership of mino

btw, I vote for Melo😉

**Comment by MangiDu** [comment]
what about 'migo', came from amigo

**Comment by adekoyejoakinhanmi** [comment]
I think Muto is fine. Although, I also liked mint.

**Comment by Shubham-Sinha** [comment]
How about Indus representing perennial source for building a civilisation. I guess your vision is that present Moon helps dev to build products around it. Secondly, you're from India, much like me. Indian devs can help contribute like the Chinese did it for Vue.

**Comment by kbrsh** [comment]
@Shubham-Sinha I think the name should be related to being fast, minimal, slim, or flexible. Also, people from anywhere around the world, any background, and any ethnicity can contribute to Moon. It's a repo where anyone (beginners to experts) can contribute and should feel welcome.

**Comment by dmjio** [comment]
I really don't recommend renaming, moon is a great name (and unique imo), and you have already gotten a lot of brand recognition for it. It is also a small name as well. You are #1 on google search for the term `moon js` as well.

**Comment by kbrsh** [comment]
@dmjio See my reasons above. Having such a general term (Moon) seems cool, but really restricts me from expanding the community. I'd like to get my own file extension for single-file components (`.moon` is taken). Another big thing is a NPM package name that isn't taken, along with a Twitter account.

**Comment by JorgeBucaran** [comment]
A few more ideas with ❤️: 

- Moonshine
- Moonshot
- Moonlight

👋

**Comment by DakshMiglani** [comment]
I vote for Melo and Moonshine, Moonshot and Moonlight are pretty big names.

**Comment by microspino** [comment]
**Mino** is a fantastic name.
I'm biased because I had a small camera with a snap out USB connector...Brilliant idea killed by Cisco. Fast, simple. *minimal UI*, ultra versatile. Loved every minute with it.
https://en-blog.files.wordpress.com/2009/11/flip_mino_hd1.png

**Comment by ElliotPsyIT** [comment]
With Moon such a close relative of Vue, seems like that association should be reflected in the name.

@seanseany 's clever suggestion of 'prevue' does this by taking a page out of the React/Preact book of a faster and slimmer approach that nods to its inspiration within its name.

Other options along the same lines might be 'minvue' or 'minvu' or 'minivu' -- starts with 'm' and acknowledges the project's influence.

**Comment by palashmon** [comment]
## My vote goes for

- `revue`
- `emu`
- `prevue`
- `one`
- `comet`

**Comment by sudomabider** [comment]
If the dependencies and all the file extensions have to change, why not the directives too?
It's an extremely easy change in any modern editor/IDE. Something that doesn't necessarily start with `m` but is somewhat related to `moon` (e.g. luna.js) will probably make the transition smoother.

I understand not having to change the directives is nice. And I like all the suggestions so far. But I don't think you have to limit yourself to an "m" word if there's a really good candidate somewhere else in the alphabet.

**Comment by kbrsh** [comment]
@ElliotPsyIT I would say Moon is a mixture of Vue, Ractive, Marko, and Svelte. I honestly don't want this to come off as a project that has API compatibility with Vue, but rather a small library that allows you to define reactive views.

@palashmon Thanks! That helps put things into perspective.

@sudomabider I agree. Starting with `m` is now not required, but I will prefer that it be four letters  and quick to type.

This will result in a change of directives, plugins, the core, component instance constructors, root instance constructors, the global API, the instance methods, and HyperMoon (Moon's virtual DOM engine).

Having `m` at the start of the name is now not a requirement everyone! I will be editing the post above to reflect that as well, but any suggestions are now welcome 👍

**Comment by hsen-dev** [comment]
Mino

**Comment by jrtnq514** [comment]
Just throwing ideas out, didn't dig too deep into them.
- `reef`
- `lune`
- `waka`
- `zoon`
- `sly`
- `rig`
- `vito` - derived from the Latin word "vita", meaning "life"

**Comment by ElliotPsyIT** [comment]
@kbrsh Good point about the cross-naming implying API compatibility in addition to a nod to inspiration.  I see now that you aren't intending to model the API - no, I haven't seen v1's API, but I'll have a look.

**Comment by palashmon** [comment]
'reef', 'lune', 'waka', 'zoon', 'sly', 'rig', 'vito'

None of them are available on npm.

**Comment by dmjio** [comment]
Still think community will grow even with moon, look at Elm for example or even react. These are generic names (neither of which has the .com or Twitter), but they've been around for a while and have large communities.

**Comment by kbrsh** [comment]
@dmjio That's not the only problem. Sure, "React" is a common term. Moon on the other hand, is an _extremely_ generic thing that is most commonly searched for by people wanting knowledge about the Earth's moon.

Along with that, I don't have the NPM name for `moon`, a valid file extension (`.moon` is already being used), and even the `@moonjs` Twitter is taken.

**Comment by emoticon** [comment]
month :peka:

**Comment by ElliotPsyIT** [comment]
Moon's intent is to create projects w/ reactive views - ie. views aware of changes around it. Perhaps consider direct names like `aware` or `awareOf`

**Comment by kokujin** [comment]
I'm for "Mino". Although I like Moon. Mino sounds "minimal" and its something one can suggest to his boss

**Comment by kbrsh** [comment]
@kokujin Yes, all related projects will be renamed (with the same policies as mentioned).

Alright everyone, here is a little poll of the best names that I think were mentioned. Click on the one you think is best.

[![](https://m131jyck4m.execute-api.us-west-2.amazonaws.com/prod/poll/01BTS77PE55TSKDKGEYMQSBPBR/Mica)](https://m131jyck4m.execute-api.us-west-2.amazonaws.com/prod/poll/01BTS77PE55TSKDKGEYMQSBPBR/Mica/vote)
[![](https://m131jyck4m.execute-api.us-west-2.amazonaws.com/prod/poll/01BTS77PE55TSKDKGEYMQSBPBR/Melo)](https://m131jyck4m.execute-api.us-west-2.amazonaws.com/prod/poll/01BTS77PE55TSKDKGEYMQSBPBR/Melo/vote)
[![](https://m131jyck4m.execute-api.us-west-2.amazonaws.com/prod/poll/01BTS77PE55TSKDKGEYMQSBPBR/Mino)](https://m131jyck4m.execute-api.us-west-2.amazonaws.com/prod/poll/01BTS77PE55TSKDKGEYMQSBPBR/Mino/vote)
[![](https://m131jyck4m.execute-api.us-west-2.amazonaws.com/prod/poll/01BTS77PE55TSKDKGEYMQSBPBR/Muto)](https://m131jyck4m.execute-api.us-west-2.amazonaws.com/prod/poll/01BTS77PE55TSKDKGEYMQSBPBR/Muto/vote)

**Comment by coderdiaz** [comment]
**Melo** likes me

**Comment by cthulberg** [comment]
A "melo" is an apple tree here in Italy.
I love it!

**Comment by microspino** [comment]
Which means @coderdiaz is liked by an 🍎  🌳  :rofl: 
@cthulberg bye from 🇮🇹

**Comment by AndreasGassmann** [comment]
I for one really like the name "moon" and hope that you will keep it.

The fact that it is an extremely generic word doesn't seem like a problem to me. Sure, if someone searches for "moon", then they will not find this library. But is that really a problem? Everyone in the js community knows about this (I think it was the same for react in the early days) and will simply add "js", which will bring moonjs to the top of the list. For me it's also easier to remember a real word rather than a fictional one. (Was it Mino? Melo? Pino?).

I understand that the file extensions and twitter handle would be nice to have. The file extension could be `.mnjs`, not ideal, but at least it sounds similar. We only need to come up with a good twitter handle.

**Comment by kbrsh** [comment]
@AndreasGassmann Glad you like it.

I definitely won't change the name if I can find a reasonable file extension and Twitter handle—changing the name requires a _huge_ amount of work. I'll need to create a new website, redo the branding, change internal variables, change all of the related modules (`moon-router`, `monx`, `moon-component-compiler`, `moon-cli`, `moon-template`, `moon-bars`), etc.

Let me know if you can come up with any plausible ideas for a file extension and Twitter handle.

**Comment by Temeez** [comment]
How about "moon" that is not "moon" but "muun"? 😅 

`muun`
`.muu`
`muun_js`

Could of course try to find a way to get a hold of existing twitter handle for "moon", but that might require even more work than renaming all the things.

**Comment by kbrsh** [comment]
@Temeez That _might work_, but I'm not fully into it.

I'm actually leaning towards keeping the name now, as I agree with @AndreasGassmann. I've gotten `moon` on NPM. All we need now is a file extension and possibly a Twitter username.

**Comment by DanielSokil** [comment]
Mova means language in Ukrainian -> мова

**Comment by oltdaniel** [comment]
@kbrsh I think it would be way easier to create a short file extension _(even when it is already used but not popular at all)_ and a Twitter Name shouldn't be the issue at all _(usernames at all are hard to find)_.

And as in Wikipedia listed ([here](https://en.m.wikipedia.org/wiki/List_of_filename_extensions_(M–R))) there is no `.moon` listed.

And I think a shorter version would make very develop happier :satisfied:.

**Comment by riiccardo** [comment]
i like `moon` simple :gem:

 just imagine the `.melo` format on files :| :scream: 

If you think about the `Moon`, two things will come to your mind: **`1- moon`**  **`2- earth`**
Suppose the `earth` is a `Vue.js` and `moon` is `Moon.js` :||||||

`moon` is lightweight than `earth` ===> :tada: `moon.js` Small file size (7kb minified + gzip) 

`moon` faster than `earth`  ===> :zap:  Blazing fast performance

`moon` orbit around `earth` ===> inspired by Vue

If you look at the sky every night, you could see the moon xdddd

**Comment by kbrsh** [comment]
@riiccardo As I said above, I'm keeping the name Moon :)

NPM Name: `moon`
File Format: `mvl`
Twitter: TBD

**Comment by dmjio** [comment]
@kbrsh great work.
